### PR TITLE
Improve title validator

### PR DESCRIPTION
**Pull request by michamos**
* Refs cern-sis/issues-inspire#23

### CODE REVIEW

File: inspire_matcher/utils.py [hunk 1]
```diff
@@ -92,3 +91,29 @@ def get_tokenized_title(title):
...
+    Returns:
+        float: a score indicating the overlap of both titles, which is ``0`` if
+            below the threshold
+    """
+    x_title_tokens = get_tokenized_title(x_title)
+    y_title_tokens = get_tokenized_title(y_title)
+    current_title_jaccard = compute_jaccard_index(x_title_tokens, y_title_tokens)
+
+    some_title_has_math = (
+        "<math>" in x_title or "$" in x_title or "<math>" in y_title or "$" in y_title
```
Comment (MJedr) [inline inspire_matcher/utils.py:115]: wouldn't regex be better for that?

**Review by MJedr — APPROVED** [review]

File: inspire_matcher/validators.py [hunk 2]
```diff
@@ -60,19 +59,15 @@ def authors_titles_validator(record, result):
...
-    title_max_score = 0.0
     record_titles = get_value(record, 'titles.title', [])
     result_titles = get_value(result, '_source.titles.title', [])
 
-    for cartesian_pair in product(record_titles, result_titles):
-        record_title_tokens = get_tokenized_title(cartesian_pair[0])
-        result_title_tokens = get_tokenized_title(cartesian_pair[1])
-        current_title_jaccard = compute_jaccard_index(record_title_tokens, result_title_tokens)
+    title_score = max(
+        compute_title_score(record_title, result_title, threshold=0.5, math_threshold=0.3)
```
Comment (MJedr) [inline inspire_matcher/validators.py:66]: Why math threshold is much lower than the normal one? Is mathml content so inconsistent that sometimes it's present and sometimes not?

File: inspire_matcher/utils.py (hunk 1 above)
Comment (michamos) [reply inspire_matcher/utils.py:115]: I don't know, escaping `$` would make it less readable and titles are short, so there's not much of a performance gain.

File: inspire_matcher/validators.py (hunk 2 above)
Comment (michamos) [reply inspire_matcher/validators.py:66]: The problem is not that math is sometimes present and sometimes not, but that we don't have a good way of normalizing math across different representations. We get either of MathML/unicode/LaTeX, where arXiv will typically use the latter and publishers one of the two others. So we get things like
`Search for the decay $D^{0} \to \pi^{0} \nu \bar{\nu}$` and `Search for the decay D0→π0νν¯` which are morally the same but don't match as they don't reach the general threshold of 0.5 for the Jaccard index.

**Pull request merged**